compiler_opt/tools/combine_tfa_policies_lib_test.py

```diff
@@ -0,0 +1,151 @@
+from tf_agents.typing import types
+import hashlib
+import numpy as np
+
+
+def client_side_model_selector_calculation(policy_name: str) -> types.Tensor:
+  m = hashlib.md5()
+  m.update(policy_name.encode('utf-8'))
+  high = int.from_bytes(m.digest()[8:], 'little')
+  low = int.from_bytes(m.digest()[:8], 'little')
+  model_selector = tf.constant([[high, low]], dtype=tf.uint64)
+  return model_selector
+
+
+class AddOnePolicy(tf_agents.policies.TFPolicy):
+  """Test policy which increments the obs feature."""
+
+  def __init__(self):
+    obs_spec = {'obs': tensor_spec.TensorSpec(shape=(1,), dtype=tf.int64)}
+    time_step_spec = ts.time_step_spec(obs_spec)
+
+    act_spec = tensor_spec.TensorSpec(shape=(1,), dtype=tf.int64)
+
+    super().__init__(time_step_spec=time_step_spec, action_spec=act_spec)
+
+  def _distribution(self, time_step):
+    """Boilerplate function for TFPolicy."""
+    pass
+
+  def _variables(self):
+    """Boilerplate function for TFPolicy."""
+    return ()
+
+  def _action(self, time_step, policy_state, seed):
+    """Boilerplate function for TFPolicy."""
+    observation = time_step.observation['obs'][0]
+    action = tf.reshape(observation + 1, (1,))
+    return policy_step.PolicyStep(action, policy_state)
+
+
+class SubtractOnePolicy(tf_agents.policies.TFPolicy):
+  """Test policy which decrements the obs feature."""
+
+  def __init__(self):
+    obs_spec = {'obs': tensor_spec.TensorSpec(shape=(1,), dtype=tf.int64)}
+    time_step_spec = ts.time_step_spec(obs_spec)
+
+    act_spec = tensor_spec.TensorSpec(shape=(1,), dtype=tf.int64)
+
+    super().__init__(time_step_spec=time_step_spec, action_spec=act_spec)
+
+  def _distribution(self, time_step):
+    """Boilerplate function for TFPolicy."""
+    pass
+
+  def _variables(self):
+    """Boilerplate function for TFPolicy."""
+    return ()
+
+  def _action(self, time_step, policy_state, seed):
+    """Boilerplate function for TFPolicy."""
+    observation = time_step.observation['obs'][0]
+    action = tf.reshape(observation - 1, (1,))
+    return policy_step.PolicyStep(action, policy_state)
+
+
+observation_spec = ts.time_step_spec({
+    'obs':
+        tf.TensorSpec(dtype=tf.int32, shape=(), name='obs'),
+    'model_selector':
+        tf.TensorSpec(shape=(2,), dtype=tf.uint64, name='model_selector')
+})
+
+action_spec = tensor_spec.TensorSpec(shape=(1,), dtype=tf.int64)
+
+
+class FeatureImportanceTest(absltest.TestCase):
+
+  def test_select_add_policy(self):
+    policy1 = AddOnePolicy()
+    policy2 = SubtractOnePolicy()
+    combined_policy = combine_tfa_policies_lib.CombinedTFPolicy(
+        tf_policies={
+            'add_one': policy1,
+            'subtract_one': policy2
+        },
+        time_step_spec=observation_spec,
+        action_spec=action_spec)
+
+    model_selector = client_side_model_selector_calculation('add_one')
+
+    state = ts.TimeStep(
+        discount=tf.constant(np.array([0.]), dtype=tf.float32),
```

Comment:
nit: can you use non-zero values for the input to the test, use like `42` :). 

0 could mask what happens if you have (1) zero-initialized values that (2) don't actually get passed to the underlying model.

Reply:
Changed the obs to 42, the other parts of the TimeStep are not used for this test.